### PR TITLE
DBZ-871 DBZ-875 Add tableId to source block

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-871
https://issues.jboss.org/browse/DBZ-875

### CODE REVIEW

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SourceInfoTest.java [hunk 1]
```diff
@@ -5,28 +5,40 @@
...
     public void beforeEach() {
         source = new SourceInfo("serverX");
         source.setChangeLsn(Lsn.NULL);
+        tableId = new TableId("c", "s", "t");
     }
 
     @Test
     public void versionIsPresent() {
-        assertThat(source.struct().getString(SourceInfo.DEBEZIUM_VERSION_KEY)).isEqualTo(Module.version());
+        assertThat(source.struct(tableId).getString(SourceInfo.DEBEZIUM_VERSION_KEY)).isEqualTo(Module.version());
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SourceInfoTest.java]: Can you add an assertion to one of the integration tests, too? It seems the s sourced from the catalog id; does this match with the configured database name for SQL Server?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java [hunk 2]
```diff
@@ -132,8 +130,6 @@
                                                      .field(BINLOG_ROW_IN_EVENT_OFFSET_KEY, Schema.INT32_SCHEMA)
                                                      .field(SNAPSHOT_KEY, SchemaBuilder.bool().optional().defaultValue(false).build())
                                                      .field(THREAD_KEY, Schema.OPTIONAL_INT64_SCHEMA)
-                                                     .field(DB_NAME_KEY, Schema.OPTIONAL_STRING_SCHEMA)
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java]: Is there any impact in terms of compatibility (when using the registry) due to the changed order? Seems it'd be safer to leave it as is?

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseSourceInfo.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    protected Struct struct(TableId tableId) {
+        final Struct sourceInfo = super.struct();
+
+        if (tableId == null) {
+            return sourceInfo;
+        }
+        if (tableId.catalog() != null) {
+            sourceInfo.put(DB_NAME_KEY, tableId.catalog());
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseSourceInfo.java]: I'm having some doubts about using the catalog. Won't it be safer to use the configured database name explicitly (as already done by some connectors at least)?

**Comment by gunnarmorling** [comment]
@jpechane, *ping*, just to make sure my comments don't go missed :)

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java (hunk 2 above)
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java]: @gunnarmorling Unfortunately you are right, for Avro it breaks backward compatibility. Can't we afford to do it in minor version change and thus postpone this to 0.10?

**Comment by gunnarmorling** [comment]
@jpechane, seems we can close this one in favour of your new one from yesterday?

**Comment by jpechane** [comment]
Superseded with https://github.com/debezium/debezium/pull/886

**Pull request closed without merging**